python/source.py

```diff
@@ -875,34 +876,74 @@ def get_fields(self, sim):
         )
         waist_xy = np.array([np.real(X0[0]), np.real(X0[1])])[:, np.newaxis, np.newaxis]
 
+        # Find large imaginary argument hankel shift
+        r, _ = self.get_r_rhat(X, X0)
+        kr = (k * r).flatten()
+        max_imag = np.abs(kr.imag[np.argmax(np.abs(kr.imag))])
+        max_float = int(np.log(float_info.max) / 2)
+        shift = 0 if np.abs(max_imag) < max_float else max_imag - max_float
+
+        # Fix hankel functions for large imaginary arguments and remove nans and infs when non-indexed
+        scaled_hankel1 = lambda o, kr: hankel1e(o, kr) * np.exp(1j * kr - shift)
+        scaled_hankel2 = lambda o, kr: hankel2e(o, kr) * np.exp(-1j * kr - shift)
+        scaled_jv = lambda o, kr: jve(o, kr) * np.exp(np.abs(kr.imag) - shift)
+
         # Get field for outgoing points (hankel2)
-        o_fields2D = self.green2d(X, freq, eps, mu, X0, kdir, hankel2, beam_E0)[
-            :, :, :, np.newaxis
-        ]
+        o_fields2D = self.green2d(
+            X[:, outgoing_arg][..., np.newaxis],
```

Comment:
I am now passing in my masks for `outgoing_arg`, `incoming_arg`, and `waist_points` before calling green2D to eliminate the overflows that were happening when the wrong hankel was being used for a given argument despite those indices of `X` not  being used. Before I was doing this while constructing the final `fields2D` array.